hack/tools/ci-install-containerd.sh

```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# script to ensure containerd binaries for e2e testing
+set -o errexit -o nounset -o pipefail
+
+# cd to repo root
+REPO_ROOT="$(cd "$(dirname "${BASH_SOURCE[0]}")/../.." &> /dev/null && pwd -P)"
+cd "${REPO_ROOT}"
+
+containerd_version='1.7.0'
+
+containerd_path="${REPO_ROOT}/bin/containerd"
+if [[ -f "${containerd_path}" ]] && "${containerd_path}" --version | grep -q "${containerd_version}"; then
+    echo "Already have ${containerd_path} ${containerd_version}"
+else
+    # downlod containerd to bindir
+    wget -qO- \
+        "https://github.com/containerd/containerd/releases/download/v1.7.0/containerd-1.7.0-linux-amd64.tar.gz" \
+    | tar -C "${REPO_ROOT}/bin" -zxvf - --strip-components=1
+fi
+
+# install pull util
+(cd "${REPO_ROOT}/hack/tools/containerd-puller" && go build -o "${REPO_ROOT}/bin/containerd-puller" .)
+
+# create a config that will let the current user run rootlessly
```

Comment:
Todo: more accurate comment here.
We're making everything the current user which is fine because we won't actually do much. If we didn't set the uid/gid it will default to chmod to root for security reasons

Reply:
here it is - https://github.com/containerd/containerd/pull/8368

Reply:
updated this comment